### PR TITLE
Add is_integer

### PR DESCRIPTION
**Pull request by rikardn**
Add a `is_integer` function.

### CODE REVIEW

File: symengine/test_visitors.cpp [hunk 1]
```diff
@@ -115,6 +115,48 @@ tribool is_nonnegative(const Basic &b)
...
+{
+    if (eq(x, *pi) or eq(x, *E) or eq(x, *EulerGamma) or eq(x, *Catalan)
+        or eq(x, *GoldenRatio)) {
+        is_integer_ = tribool::trifalse;
+    } else {
+        is_integer_ = tribool::indeterminate;
+    }
+}
+
+void IntegerVisitor::bvisit(const Add &x)
```
Comment (isuruf) [inline symengine/test_visitors.cpp:137]: Do the same for Mul?

Comment (rikardn) [reply symengine/test_visitors.cpp:137]: Sure!

**Review by isuruf — COMMENTED** [review]
Looks good. We can add a few methods like `Pow, Abs, Floor, Ceiling`, but it's not necessary for this PR

**Comment by rikardn** [comment]
Thanks @isuruf for reviewing! Agree, I will probably continue adding more to this in other PRs.

**Pull request merged**